src/algebra/squarefree.lean

```diff
@@ -204,4 +204,39 @@ lemma sum_divisors_filter_squarefree {n : ℕ} (h0 : n ≠ 0)
 by rw [finset.sum_eq_multiset_sum, divisors_filter_squarefree h0, multiset.map_map,
     finset.sum_eq_multiset_sum]
 
+lemma sq_mul_squarefree {n : ℕ} (hn : 0 < n) :
+  ∃ a b : ℕ, 0 < a ∧ 0 < b ∧ b ^ 2 * a = n ∧ squarefree a :=
```

Comment:
Why have the `0 < n` hypothesis? Couldn't you just drop `0 < a` (and possibly also `0 < b`) in the conclusion?

(An alternative would be to include uniqueness in the statement, where I do see the value of including positivity.)

Reply:
Oh, I see, you wouldn't have `square_free 0`. I guess it's fine as is.

Reply:
I guess I could change that lemma, because 0 = 0^2 * a, for any a. And then deduce positivity of a and b from the positivity of n in the sq_mul_squarefree'.

Reply:
I actually had to keep the lemma for just positive n, and include a new lemma that doesn't need the 0 < n hypothesis, but just splits into case n = 0 and case n > 0.